Lib/argparse.py

```diff
@@ -1697,6 +1697,28 @@ def add_mutually_exclusive_group(self, *args, **kwargs):
         return super().add_mutually_exclusive_group(*args, **kwargs)
 
 
+def _prog_name(prog=None):
+    if prog is not None:
+        return prog
+    arg0 = _sys.argv[0]
+    try:
+        modspec = getattr(_sys.modules['__main__'], "__spec__", None)
+    except KeyError:
+        # possibly PYTHONSTARTUP or -X presite
+        # no good answer here, so fall back to the default
+        modspec = None
```

Comment:
Alternatively, we could use `getattr(_sys.modules.get('__main__'), '__spec__', None)` or catch AttributeError together with KeyError to emphasize that these cases are similar.

The latter is more preferable because allows to handle these cases differently.
```suggestion
    try:
        modspec = _sys.modules['__main__'].__spec__
    except (KeyError, AttributeError):
        # possibly PYTHONSTARTUP or -X presite or other weird edge case
        # no good answer here, so fall back to the default
        modspec = None
```